packages/elements/src/internals/machines/sign-in.machine.ts

```diff
@@ -72,55 +76,94 @@ export const SignInMachine = setup({
   },
   types: {
     context: {} as {
-      client: SignInClient;
+      clerk: LoadedClerk;
       router: ClerkHostRouter;
       error?: Error | ClerkAPIResponseError;
       resource?: SignInResource;
-      fields: Record<string, { error: string }>;
+      fields: Record<string, { value: string; error: string }>;
     },
     input: {} as {
-      client: SignInClient;
+      clerk: LoadedClerk;
       router: ClerkHostRouter;
     },
-    events: {} as { type: 'START' } | { type: 'SUBMIT' } | { type: 'NEXT' } | { type: 'RETRY' } | { type: 'ASSIGN' },
+    events: {} as
+      | { type: 'START' }
+      | { type: 'SUBMIT'; fields: Record<string, any> }
+      | { type: 'NEXT' }
+      | { type: 'RETRY' }
+      | { type: 'ASSIGN' },
   },
 }).createMachine({
   context: ({ input }) => ({
-    client: input.client,
+    clerk: input.clerk,
     router: input.router,
     currentFactor: null,
     fields: {},
   }),
   initial: STATES.Init,
   states: {
     [STATES.Init]: {
-      always: 'Start',
+      invoke: {
+        src: 'waitForClerk',
+        input: ({ context }) => context.clerk,
+        onDone: {
+          target: STATES.Start,
+          actions: assign({
+            // @ts-expect-error -- this is really IsomorphicClerk up to this point
+            clerk: ({ context }) => context.clerk.clerkjs,
+          }),
+        },
+      },
     },
     [STATES.Start]: {
       entry: ({ context }) => console.log('Start entry: ', context),
       on: {
-        SUBMIT: STATES.StartAttempting,
+        SUBMIT: {
+          actions: assign({
+            fields: ({ event }) => event.fields,
+          }),
+          target: STATES.StartAttempting,
+        },
       },
     },
     [STATES.StartAttempting]: {
       entry: ({ context }) => console.log('StartAttempting entry: ', context),
       invoke: {
-        id: 'createSignIn',
         src: 'createSignIn',
-        input: ({ context }) => ({
-          client: context.client,
-          params: {
-            identifier: 'tom@clerk.dev',
-            password: 'tom@clerk.dev',
-            strategy: 'password',
-          },
-        }),
-        onDone: [{ actions: 'assignResourceToContext' }],
+        input: ({ context }) => {
```

Comment:
❓ @BRKalow Thinking about this, might we want to pass all fields and parse in `createSignIn`? Not at all blocker, but food for thought.

Reply:
works for me! It also keeps the logic out of the machine 😄